test/fixtures/hdfs-fixture/build.gradle

```diff
@@ -18,25 +18,22 @@
  */
 
 apply plugin: 'elasticsearch.build'
+apply plugin: 'elasticsearch.test.fixtures'
 
-versions << [
-  'hadoop2': '2.8.1'
-]
-
-// we create MiniHdfsCluster with the hadoop artifact
 dependencies {
-  compile "org.apache.hadoop:hadoop-minicluster:${versions.hadoop2}"
+  compile "org.apache.hadoop:hadoop-minicluster:2.8.1"
+}
+
+task syncClasses(type: Sync) {
+  from sourceSets.test.compileClasspath
+  from sourceSets.test.output.dirs
+  into "${buildDir}/fixture"
+  dependsOn sourceSets.test.getCompileTaskName("java")
 }
 
-// for testing, until fixtures are actually debuggable.
-// gradle hides *EVERYTHING* so you have no clue what went wrong.
-task hdfs(type: JavaExec) {
-  classpath = sourceSets.test.compileClasspath + sourceSets.test.output
-  main = "hdfs.MiniHDFS"
-  args = [ 'build/fixtures/hdfsFixture' ]
+preProcessFixture.dependsOn syncClasses
```

Comment:
can we combine these two lines under a `preProcessFixture { }` block. If nothing else I find config blocks visual break up the script and make these a bit easier to read.